sphinx/util/inspect.py

```diff
@@ -158,3 +158,26 @@ def is_builtin_class_method(obj, attr_name):
     if not hasattr(builtins, safe_getattr(cls, '__name__', '')):
         return False
     return getattr(builtins, safe_getattr(cls, '__name__', '')) is cls
+
+
+def dumb_signature(functional_object):
+    src = inspect.getsourcelines(functional_object)[0]
+
+    for i, l in enumerate(src):
+        if 'def' in l:
+            break
+    acc = ''
+    for l in src[i:]:
+        acc += l.strip()
+        if '):' in acc:
+            break
+
+    r = re.compile(r'\((.*)\)')
+    s = r.findall(acc)
+    if not s:
+        return '()'
+    s = s[0]
+    for d in [r'^self,', r'^self', r'^cls,', r'^cls']:
```

Comment:
This introduces new behaviour where ```def foo(self, cls, bar)``` is stripped down to `(bar)` where, without this setting, `cls` would be retained.

Also, IIUC it would erroneously strip *any* parameter prefixed with `self`/`cls` currently, eg. ```def bar(selfevident_parameter):```.